baseapp/abci.go

```diff
@@ -248,11 +248,15 @@ func (app *BaseApp) EndBlock(req abci.RequestEndBlock) (res abci.ResponseEndBloc
 // Ref: https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-060-abci-1.0.md
 // Ref: https://github.com/tendermint/tendermint/blob/main/spec/abci/abci%2B%2B_basic_concepts.md
 func (app *BaseApp) PrepareProposal(req abci.RequestPrepareProposal) (resp abci.ResponsePrepareProposal) {
-	ctx := app.getContextForTx(runTxPrepareProposal, []byte{})
 	if app.prepareProposal == nil {
 		panic("PrepareProposal method not set")
 	}
 
+	ctx := app.getContextForTx(runTxPrepareProposal, []byte{}).
+		WithBlockHeight(req.Height).
```

Comment:
This seems to be set to a valid value after the first `PrepareProposal` call. Do you know why this might be happening?

Reply:
Late response, but this was happening because you would be seeing the previous block height instead of the current. So it gave the illusion of being a valid value